### PR TITLE
feat: adding full text search support

### PR DESCRIPTION
**Pull request by gsi-alejandro**
Closes: #745

### CODE REVIEW

File: __test__/fts.spec.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { searchQuery, SearchQuery } from '../src';
+
+const maybe = process.env.CI ? test.skip : test;
```
Comment (ejscribner) [inline __test__/fts.spec.ts:3]: Does this mean we skip these tests on CI? Is that due to the cb server version not supporting FTS?

File: docusaurus/docs/advanced/fts.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+## Scan Consistency and ConsistentWith
+
+By default, all Search queries will return the data from whatever is in the index at the time of query. 
+These semantics can be tuned if needed so that the hits returned include the most recently performed mutations,
+at the cost of slightly higher latency since the index needs to be updated first.
+
+There are two ways to control consistency: either by supplying a custom `SearchScanConsistency` or using `consistentWith`.
+At the moment the cluster only supports `consistentWith`, which is why you only see `SearchScanConsistency.NotBounded`
+in the enum which is the default setting.
+The way to make sure that recently written documents show up in the search works as follows (commonly referred to "`read your own writes"` — RYOW):
```
Comment (ejscribner) [inline docusaurus/docs/advanced/fts.md:191]:
```suggestion
The way to make sure that recently written documents show up in the search works as follows (commonly referred to as **"read your own writes"**  or **RYOW**):
```

Comment (ejscribner) [reply docusaurus/docs/advanced/fts.md:191]: Not really sure if this makes sense but wasn't sure if it should be a codeblock

File: docusaurus/docs/advanced/fts.md
Comment (ejscribner) [inline docusaurus/docs/advanced/fts.md:1]: This docs page looks really good 🙂 nice work!

**Comment by ejscribner** [comment]
This all looks awesome! Just a few small nits, but otherwise looks great!

File: __test__/fts.spec.ts (hunk 1 above)
Comment (gsi-alejandro) [reply __test__/fts.spec.ts:3]: Tests were marked to be skipped in the CI due to the data being stored in an FTS index and we don't know the time for they to be ready and return the correct values, so we tried adding some delays but it is uncertain and only worked a few times with 15 sec of delay which is too much delay and even with that fails most the times.

Note: Testing indexed data in a CI can be tricky

File: docusaurus/docs/advanced/fts.md (hunk 2 above)
Comment (gsi-alejandro) [reply docusaurus/docs/advanced/fts.md:191]: you're correct, the fix will be to remove the ` characters

**Review by ejscribner — APPROVED** [review]

**Pull request merged**